src/js/GoldifyExecutePage.js

```diff
@@ -1,14 +1,196 @@
-import React, { Component } from 'react';
+import React, { Component } from "react";
 import "../css/GoldifyExecutePage.css";
 
+const queryString = require("query-string");
+const goldifyUtils = require("../utils/GoldifyUtils");
+const goldifyTestUtils = require("../utils/GoldifyTestUtils");
+
+var clientId = process.env.REACT_APP_SPOTIFY_CLIENT_ID; // Your client id
+var clientSecret = process.env.REACT_APP_SPOTIFY_CLIENT_SECRET; // Your secret
+var redirectUri = process.env.REACT_APP_SPOTIFY_REDIRECT_URI; // Your redirect uri
+
+/**
+ * Generates a random string containing numbers and letters
+ * @param  {number} length The length of the string
+ * @return {string} The generated string
+ */
+var generateRandomString = function(length) {
+  var text = '';
+  var possible = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+
+  for (var i = 0; i < length; i++) {
+    text += possible.charAt(Math.floor(Math.random() * possible.length));
+  }
+  return text;
+};
+
+function useQuery() {
+  return new URLSearchParams(window.location.search);
+}
+
+function retrieveAuthorization() {
+  let spotifyApiURL =
+    "https://accounts.spotify.com/authorize?" +
+    queryString.stringify({
+      response_type: "code",
+      client_id: clientId,
+      scope: "user-read-private user-read-email",
+      redirect_uri: redirectUri,
+      state: generateRandomString(16)
+    });
+  replaceWindowURL(spotifyApiURL);
+}
+
+function replaceWindowURL(url) {
+  if (!goldifyUtils.jestTestIsRunning()) {
```

Comment:
I feel weird about adding logic only related to if tests are running - this is fine for an initial go but i'd really like if we can get rid of this somehow, and maybe mock certain functions if we need in the tests themselves

Reply:
This is a great point, and not the right move by me. Could get very sloppy.

For record keeping, this has been resolved by using mocking in our test suites.